pkg/users/group-details.js

```diff
@@ -0,0 +1,185 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import cockpit from 'cockpit';
+import React, { useState } from 'react';
+import { superuser } from "superuser";
+import { apply_modal_dialog } from "cockpit-components-dialog.jsx";
+import { show_unexpected_error } from "./dialog-utils.js";
+
+import {
+    Button,
+    Card, CardBody, CardHeader, CardTitle, CardActions,
+    Flex, HelperText, HelperTextItem,
+    Label, LabelGroup,
+    Page, PageSection,
+    Select, SelectOption, SelectVariant, Text, TextVariants, Breadcrumb, BreadcrumbItem,
+    Form, FormGroup,
+} from '@patternfly/react-core';
+import { UndoIcon } from '@patternfly/react-icons';
+
+import { GroupActions } from "./group-actions.jsx";
+
+const _ = cockpit.gettext;
+
+export function GroupDetails({ groups, accounts, group }) {
+    const current_group = groups.find(g => g.name == group);
+    return (
+        <Page groupProps={{ sticky: 'top' }}
+              isBreadcrumbGrouped
+              id="group"
+              breadcrumb={
+                  <Breadcrumb>
+                      <BreadcrumbItem to="#/">{_("Groups")}</BreadcrumbItem>
+                      <BreadcrumbItem isActive>{group}</BreadcrumbItem>
+                  </Breadcrumb>}>
+            <PageSection>
+                <Card className="group-details" id="group-details">
+                    <CardHeader>
+                        <CardTitle id="group-title"><Text component={TextVariants.h2}>{group}</Text></CardTitle>
+                        {superuser.allowed && <CardActions>
+                            <GroupActions group={current_group} accounts={accounts} />
+                        </CardActions>}
+                    </CardHeader>
+                    <CardBody>
+                        <Form isHorizontal onSubmit={apply_modal_dialog}>
+                            <FormGroup fieldId="group-name" hasNoPaddingTop label={_("Name")}>
+                                <output id="group-name">{group}</output>
+                            </FormGroup>
+                            <GroupAccountsSelect accounts={accounts} group={group} groups={groups} />
+                        </Form>
+                    </CardBody>
+                </Card>
+            </PageSection>
+        </Page>
+    );
+}
+
+export const GroupAccountsSelect = ({ accounts, group, groups }) => {
+    const current_group = groups.find(g => g.name == group);
+    const [isOpenAccount, setIsOpenAccount] = useState(false);
+    const [selected, setSelected] = useState([...current_group.userlist, ...current_group.userlistPrimary]);
+    const [history, setHistory] = useState([]);
+    const [loggedInUsersAffected, setLoggedInUsersAffected] = useState(false);
+
+    const undoAccountChanges = () => {
```

Comment:
This feels a bit out of place, as we don't have "undo" anywhere else in Cockpit. But it feels nice!

That said, it's easy to confuse: I added "admin" to "builder", then removed it again on the command line with `usermod -G admin admin` (which doesn't change the UI)

![image](https://user-images.githubusercontent.com/200109/222062820-5261fab9-c3c0-496d-9cb2-72c6649f4081.png)

then trying to undo causes this:

![image](https://user-images.githubusercontent.com/200109/222062893-45ae4731-e568-43a4-8f3e-af606cdfda41.png)

I think it's okay, it's a corner case and the page does not crash. A reload gets it back into shape. I'm just saying that introducing "undo" functionality in Cockpit will always have this fundamental problem.